server/src/main/java/io/crate/execution/engine/collect/DocValuesAggregates.java

```diff
@@ -134,20 +134,51 @@
 
     @Nullable
     private static MappedFieldType resolveInputToFieldType(FieldTypeLookup fieldTypeLookup,
-                                                           List<Symbol> toCollect,
                                                            Symbol input) {
-        if (!(input instanceof InputColumn)) {
+        MappedFieldType mappedFieldType = fieldTypeLookup.get(((Reference) input).column().fqn());
```

Comment:
To make it possible to get a doc values aggregator for cases like `sum(long_column::numeric)`.
The aggregation, in this case, will be a `cast` function symbol with the `numeric` value type,
so it would not be possible to resolve [DocValueAggregator](https://github.com/crate/crate/pull/10854/files#diff-f655e047f75c5f53c3c464715b73f94c02b3fa19f62209ea606a4f3d9cfec7d8L196) later on because we would try to get it for the wrong type that is not in the list of `to collect` symbols. Therefore, if we face the cast function we have to visit its argument and apply the logic we had before, checking if it is a reference or input column symbol. See https://github.com/crate/crate/pull/10854/files#diff-f655e047f75c5f53c3c464715b73f94c02b3fa19f62209ea606a4f3d9cfec7d8R163-R169

Reply:
Ah I see, function traversal is needed, but isn't this already required on the master when doing something like `sum(long_column::int)`? Looks like a bugfix, not?

Other than that, I still think that the logic is a bit shaky as with the new `ReferenceOrInputColumnSymbolExtractor` theoretically a `InputColumn` is returned while this function `resolveInputToFieldType` is requiring a `Reference`.
If we expect the return type to be a `Reference` (which is afaik always the case) we should make that clear inside the code and fail on everything else.

Reply:
> Ah I see, function traversal is needed, but isn't this already required on the master when doing something like sum(long_column::int)? Looks like a bugfix, not?

You are right, it is the case on the master already. Maybe I'll extract this part and make a PR against master. 

> Other than that, I still think that the logic is a bit shaky as with the new ReferenceOrInputColumnSymbolExtractor theoretically a InputColumn is returned while this function resolveInputToFieldType is requiring a Reference.

Agree 👍  will refine that

Reply:
> Ah I see, function traversal is needed, but isn't this already required on the master when doing something like sum(long_column::int)? Looks like a bugfix, not?

it is not a bug on, master. this case is not implemented, I'll add just a support it for the numeric cast only. I'll try to follow up on it on this week.